### ISSUE TITLE
Gap while Stage Manager ON and only 1 window opened

### ISSUE THREAD

**Issue by eshpakovsky** [rxhanson/Rectangle#1251]
When Stage Manager is turned on and only 1 window opened Rectangle still make a gap for Stage Manager while MacOS hides SM.

macOS version: 13.5.1
Rectangle version: 0.70 (76)
Logs if applicable (In Rectangle menu, hold option, "View Logging..."):

**Comment by rxhanson** [comment]
Thanks for reporting. I can't reproduce it. Could you include the logs for when this happens?

**Comment by eshpakovsky** [comment]
2023-09-07T00:18:12+03:00: AX sizing proposed: (2300.0, 1399.0), result: (2300.0, 1399.0)
2023-09-07T00:18:12+03:00: AX position proposed: (72.0, 33.0), result: (72.0, 33.0)
2023-09-07T00:18:12+03:00: AX sizing proposed: (2300.0, 1399.0), result: (2300.0, 1399.0)
2023-09-07T00:18:12+03:00: maximize | display: (64.0, 0.0, 2316.0, 1415.0), calculatedRect: (72.0, 33.0, 2300.0, 1399.0), resultRect: (72.0, 33.0, 2300.0, 1399.0), srcScreen: LG ULTRAFINE, destScreen: LG ULTRAFINE, resultScreen: LG ULTRAFINE

**Comment by rxhanson** [comment]
Hmmm, let me try this out on some other setups and get back to you.

**Comment by dch09** [comment]
Same issue here. I was thinking about adding another shortcut for maximizing window ignoring of stage manager safe area.

**Comment by rxhanson** [comment]
@matthewess thanks for looking into it. Indeed, you are on the right track with looking at `isStageStripVisible`, and I don't see anything wrong in your notes there. I was testing on a different Mac, and was able to reproduce this, but it doesn't necessarily appear to be tied to different Spaces. I'll have to do some further debugging to get a better understanding of what's going on here. 

@decodism, let me know if you want to take a look.

**Comment by rxhanson** [comment]
@matthewess, I still haven't had time to dive into this but will let you know when I do. 

For troubleshooting what's going on, my suggestion is to print off the WindowInfo for all of the windows that the getWindowList call gives back. While doing that, you can filter out windows that are in the menu bar by looking at their size, and this should give you a better picture of what's being returned.

I don't recall this giving back windows on other spaces, and in my limited testing that I've done, the issue persists regardless of what's on another space or not, so I think this might not be our concern with this issue.

**Comment by aayushbhatia06** [comment]
I am commenting here today because I just updated to 0.74 and was really disappointed to see this was reverted. When I installed 0.73 I thought this was a feature because its super annoying when I have one window open, I maximize it and it takes the full screen; and then when I open another window I first maximize the new window and THEN have to go back to the old window and "maximize" it again so it shrinks in width and stage manager is visible again. 

I understand how the behaviour in 0.73 might be annoying to people who work with single windows a lot (which kinda defeats the point of having stage manager on but different stokes for different folks I understand) but people who work with multiple windows (When I have to use a single window and simply full screen it using the green maximize button) "maximizing" two windows over and over gets old really fast. 

I think the best way would be to make it an option, if thats possible.
